libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -1006,21 +1008,27 @@ public XContentBuilder copyCurrentStructure(XContentParser parser) throws IOExce
 
     /**
      * Sets a version used for serialising a response compatible with a previous version.
+     * @param compatibleVersion - indicates requested a version of API that the builder will be creating
      */
-    public XContentBuilder withCompatibleMajorVersion(byte compatibleMajorVersion) {
-        assert this.compatibleMajorVersion == 0 : "Compatible version has already been set";
-        if (compatibleMajorVersion == 0) {
-            throw new IllegalArgumentException("Compatible major version must not be equal to 0");
+    public XContentBuilder withCompatibleVersion(CompatibleVersion compatibleVersion) {
+        assert this.restCompatibilityVersion == null : "Compatible version has already been set";
+        if (compatibleVersion == null) {
+            throw new IllegalArgumentException("Compatible major version must not be null");
```

Comment:
nit: the message needs to be updated. Also, can probably use Objects.requireNonNull